### PR TITLE
bpo-42378: fixed log truncation on logging shutdown

### PR DESCRIPTION
**Pull request by akulakov**
https://bugs.python.org/issue42378

Automerge-Triggered-By: GH:vsajip

### CODE REVIEW

**Review by vsajip — CHANGES_REQUESTED** [review]
Generally looks OK, but I would suggest the following changes:

* `_closed = True` is done in `Handler.close()`, and so there is a reliance that `FileHandler.close()` will call the superclass' `close()`. Of course it does, and there is a comment there about bpo-19523 and avoiding a handler leak, but I would add under that something  like
  ```
  # See also bpo-42378: we also rely on
  # self._closed being set to True there
  ```
* As you've added `next_rec()` to `BaseFileTest`, it can be removed from `RotatingFileHandlerTest`, which inherits from `BaseFileTest`.
* And, of course, add the documentation update and NEWS entry, as you've mentioned on the issue,

Thanks!

**Comment by akulakov** [comment]
@vsajip Thanks for the review! 
I pushed an update per comments, PTAL..

File: Doc/library/logging.handlers.rst [hunk 1]
```diff
@@ -117,6 +117,9 @@ sends logging output to a disk file.  It inherits the output functionality from
 
       Outputs the record to the file.
 
+      Note that if the file was closed due to logging shutdown at exit and file
+      mode is 'w', record will not be emitted (see :issue:`42378`).
```
Comment (vsajip) [inline Doc/library/logging.handlers.rst]: I would change "file mode" to "the file mode" and "record" to "the record".

**Review by vsajip — CHANGES_REQUESTED** [review]
Just minor changes suggested to the documentation part. Thanks!

**Comment by akulakov** [comment]
@vsajip fixed the docs per comment..

**Review by vsajip — APPROVED** [review]

**Comment by akulakov** [comment]
thanks @vsajip !

**Pull request merged**